network/src/main/java/org/jboss/as/network/OutboundSocketBinding.java

```diff
@@ -215,21 +215,21 @@ public InetAddress getOptionalSourceAddress() {
      * this outbound socket binding has a port offset. To get the absolute source port, use the {@link #getAbsoluteSourcePort()}
      * method.
      *
-     * @return the source port for this outbound socket binding
+     * @return the source port for this outbound socket binding not accounting for port offset/fixation; {@code null} if an ephemeral port should be used
      */
     public Integer getSourcePort() {
-        return this.sourcePort;
+        return sourcePort == 0 ? null : this.sourcePort;
```

Comment:
Plus let's not mix the code style; use 'this.sourceport' or just 'sourceport' consistently in the statement.

But the NPE is more important. :D